### PR TITLE
Feature/directional lists

### PR DESCRIPTION
**Pull request by Klynger**
#### What is the purpose of this pull request?
Create the directional lists in order to implement different designs of menu.
The current configuration is the following:

```
menu -> menu-item.root -> submenu -> horizontal-list -> vertical-list -> menu-item
```

For now, this is like the default configuration of the menu, but in the PR I will create a mocked hook to generate different configurations to each menu and also improve the navigation tree

#### How should this be manually tested?
https://newmenu--storecomponents.myvtex.com/ click on `more` in the header

#### Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

**Comment by Klynger** [comment]
> What is this "More"?
> image

More is the title of the menu, going to change this value to make it more clear

File: react/MenuItem.tsx [hunk 1]
```diff
@@ -1,5 +1,31 @@
...
+
+  if (!hasChildren && !label) {
+    return null
+  }
+
+  return (
+    /* This navigationItem.link might be undefined, but thats ok
+     * https://stackoverflow.com/a/10510353
+     */
+    <a className={linkClasses} href={link}>
```
Comment (klzns) [inline react/MenuItem.tsx:26]: Why not use the Link component?

File: react/Submenu.tsx [hunk 2]
```diff
@@ -49,9 +38,21 @@ export default function Submenu(props: SubmenuProps) {
...
+
   return (
     <BaseOverlay classes={overlayClasses}>
-      {items}
+      {navigation?.title && (
+        <div className={titleContainerClasses}>
+          {/* eslint-disable-next-line @typescript-eslint/no-non-null-assertion */}
+          <span className={titleClasses}>{navigation!.title}</span>
+        </div>
+      )}
```
Comment (klzns) [inline react/Submenu.tsx:54]: I think that doesnt make much sense

**Comment by klzns** [comment]
Btw, tests are failing

File: react/MenuItem.tsx (hunk 1 above)
Comment (Klynger) [reply react/MenuItem.tsx:26]: I'll do it later, for now, I just want to implement the interactions between the `SubmenuItemVerticalList` and `SubmenuItemHorizontalList`. After that, I'll fix all these problems about the wrong components/elements that are being used around these new implementation

**Pull request merged**